app/controllers/discourse_reactions/custom_reactions_controller.rb

```diff
@@ -39,6 +42,41 @@ def reactions_received
       render_serialized reaction_users.to_a, UserReactionSerializer
     end
 
+    def post_reactions_users
+      post_id = params.require(:post_id).to_i
+      reaction_value = params[:reaction_value]
+
+      post = Post.find_by(id: post_id)
+
+      raise Discourse::InvalidParameters if !post || (reaction_value && !DiscourseReactions::Reaction.valid_reactions.include?(reaction_value))
+
+      reaction_users = {}
+
+      likes = post.post_actions.where("deleted_at IS NULL AND post_action_type_id = ?", PostActionType.types[:like]) if !reaction_value || reaction_value == DiscourseReactions::Reaction.main_reaction_id
+
+      like_users = {
+        users: likes.includes([:user]).limit(MAX_USERS_COUNT + 1).map { |like| { username: like.user.username, name: like.user.name, avatar_template: like.user.avatar_template, can_undo: guardian.can_delete_post_action?(like) } },
+      } if !likes.blank?
+
+      reaction_users[DiscourseReactions::Reaction.main_reaction_id] = like_users if like_users
+
+      if !reaction_value
+        post.reactions.select { |reaction| reaction[:reaction_users_count] }.each do |reaction|
+          reaction_users[reaction.reaction_value] = {
+            users: reaction.reaction_users.includes(:user).order("discourse_reactions_reaction_users.created_at desc").limit(MAX_USERS_COUNT + 1).map { |reaction_user| { username: reaction_user.user.username, name: reaction_user.user.name, avatar_template: reaction_user.user.avatar_template, can_undo: reaction_user.can_undo? } }
```

Comment:
This block is exactly the same than the other one and has lots of logic, share it